daemons/fenced/fenced_remote.c

```diff
@@ -2243,39 +2215,42 @@ fenced_process_fencing_reply(xmlNode *msg)
         /* Could be for an event that began before we started */
         /* TODO: Record the op for later querying */
         crm_info("Received peer result of unknown or expired operation %s", id);
-        goto done;
+        pcmk__reset_result(&result);
+        return;
     }
 
+    op->result = result; // The operation takes ownership of the result
```

Comment:
Should we reset `op->result` first to be safe (now and in the future)?

Reply:
or have pcmk__take_result_ownership(&result_dest, &result_src) that takes car of this as well.

Reply:
probably a good idea to reset it first just in case

Reply:
Looks good to me other than that

Reply:
added commit on top